### ISSUE TITLE
Error installing without binary packages

### ISSUE THREAD

**Issue by atodorov** [weiwei/junitparser#83]
As part of CI testing for another project we install the resulting src and wheel packages separately without mixing them, for example

`pip install --no-binary :all: dist/kiwitcms-junit.xml-plugin*.tar.gz` or `pip install --only-binary :all: dist/kiwitcms_junit.xml_plugin*.whl`.  Previously this used to work, however upgrading to junitparser 2.4.1 breaks it. 

`junitparser` is a direct dependency of the package under test.

Steps to reproduce:

Inside a fresh venv:

```
$ pip install -U pip future setuptools
Requirement already satisfied: pip in /tmp/xml/lib/python3.8/site-packages (21.3.1)
Requirement already satisfied: future in /tmp/xml/lib/python3.8/site-packages (0.18.2)
Requirement already satisfied: setuptools in /tmp/xml/lib/python3.8/site-packages (60.2.0)

$ pip freeze
future==0.18.2

$ pip install --no-binary :all: junitparser==2.4.1
Collecting junitparser==2.4.1
  Using cached junitparser-2.4.1.tar.gz (12 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /tmp/xml/bin/python /tmp/xml/lib64/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpro0v4ray
       cwd: /tmp/pip-install-_1yhp9kd/junitparser_8c82650a77c04649bbbaf69aae6be051
  Complete output (16 lines):
  Traceback (most recent call last):
    File "/tmp/xml/lib64/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/tmp/xml/lib64/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/tmp/xml/lib64/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 130, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-qrj96qbf/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 162, in get_requires_for_build_wheel
      return self._get_build_requires(
    File "/tmp/pip-build-env-qrj96qbf/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 143, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-qrj96qbf/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 158, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 4, in <module>
      from junitparser import version
  ModuleNotFoundError: No module named 'junitparser'
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d5/ac/4461949cc80116bed8d59746b28fdf0e19717f6b69e4d81af25763fb8da1/junitparser-2.4.1.tar.gz#sha256=9bf4f474d21621072d12c174012e5e79c599fa1758087680e96793786b1dc74c (from https://pypi.org/simple/junitparser/). Command errored out with exit status 1: /tmp/xml/bin/python /tmp/xml/lib64/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py get_requires_for_build_wheel /tmp/tmpro0v4ray Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement junitparser==2.4.1 (from versions: 0.1, 0.2, 0.3, 0.4, 0.5, 0.7, 0.8, 0.9, 1.0.0, 1.2.0, 1.2.1, 1.2.2, 1.3.0, 1.3.2, 1.3.3, 1.3.4, 1.4.0, 1.4.1, 1.4.2, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.6.3, 2.0.0b1, 2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.3.0, 2.4.0, 2.4.1)
ERROR: No matching distribution found for junitparser==2.4.1
```

It is https://github.com/weiwei/junitparser/commit/fb0b9f361d649b2adcbaa9651a8cc2de4b316f35 which triggers the failure but I think the reason is that `setup.py` tries to import from `junitparser` and for some reason this isn't found. 

@weiwei my first question is what commands do you use to produce the tar.gz and wheel versions of this package ?

**Comment by weiwei** [comment]
@atodorov I use an existing github action for it. It uses `twine` for the job. I don't know about the internals though.

**Comment by weiwei** [comment]
@atodorov  Please try the fix in version 2.4.2. It works on my local machine now.

**Comment by atodorov** [comment]
@weiwei 2.4.2 works for me.
